src/cmd/disas/main.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2015 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+
+	"cmd/internal/goobj"
+	"cmd/internal/obj"
+)
+
+func objdumpcmd(fname string) *exec.Cmd {
+	GOARCH := obj.Getgoarch()
+	switch GOARCH {
+	case "arm":
+		return exec.Command(
+			"arm-none-eabi-objdump",
+			"-b", "binary",
+			"-m", "arm",
+			"-EL",
+			"-D", fname)
+	case "riscv":
+		return exec.Command(
+			"riscv64-unknown-elf-objdump",
+			"-b", "binary",
+			"-m", "riscv:rv64",
+			"-EL",
+			"-D", fname)
+	default:
+		log.Fatalf("unsupported architecture %s", GOARCH)
+	}
+	return nil
+}
+
+func disas1(sym *goobj.Sym, data []byte) {
```

Comment:
If these wrote to an io.Writer and had the temp file creation one level higher, it would be easy to have this support disassembling to files, stdout/stderr, etc...